src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,145 @@
+    private static final String TITLE = "title";
+    private static final String PRICE = "price";
+
+    @Override
+    public Book create(Book book) {
+        String sql = "INSERT INTO books (title, price) VALUES (?, ?)";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection
+                        .prepareStatement(sql, Statement.RETURN_GENERATED_KEYS)) {
+
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+
+            int affectedRow = statement.executeUpdate();
+            if (affectedRow < 1) {
+                throw new RuntimeException(
+                        "Expected to insert at leas one row, but inserted 0 rows"
+                );
+            }
+
+            ResultSet generateKeys = statement.getGeneratedKeys();
+            if (generateKeys.next()) {
+                Long id = generateKeys.getObject(1, Long.class);
+                book.setId(id);
+            }
+
+        } catch (SQLException e) {
+            throw new DataProcessingException("Cannot add new book", e);
+        }
+
+        return book;
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        String sql = "SELECT id, title, price FROM books WHERE id = ?";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(sql)) {
+
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+
+            if (resultSet.next()) {
+                Long bookId = resultSet.getLong(ID);
+                String title = resultSet.getString(TITLE);
+                BigDecimal price = resultSet.getBigDecimal(PRICE);
+
+                Book book = new Book(bookId, title, price);
+
+                return Optional.of(book);
+            } else {
+                return Optional.empty();
+            }
+
+        } catch (SQLException e) {
+            throw new DataProcessingException("Error finding book by id: " + id, e);
+        }
+    }
+
+    @Override
+    public List<Book> findAll() {
+        List<Book> books = new ArrayList<>();
+        String sql = "SELECT id, title, price FROM books";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                Statement statement = connection.createStatement();
+                ResultSet resultSet = statement.executeQuery(sql)) {
+
+            while (resultSet.next()) {
+                Long bookId = resultSet.getLong(ID);
+                String title = resultSet.getString(TITLE);
+                BigDecimal price = resultSet.getBigDecimal(PRICE);
+
+                Book book = new Book(bookId, title, price);
+
+                books.add(book);
+            }
+
+        } catch (SQLException e) {
+            throw new DataProcessingException("Error fetching all books", e);
```

Comment:
Make similar error messages, sometimes you write error, sometimes failure, sometimes can not, so please decide